### PR TITLE
Feature/vertrek punt

### PR DESCRIPTION
**Pull request by Jeffrey-Zutt**
**First version of a `DefaultModal`**

Adds the first version of a reusable modal component.
Usage:
```
<DefaultModal>
   <b>Place your components here.</b>
</DefaultModal>
```

**First version of a `AddAddressModal`**

- Is based off the `DefaultModal`
- Is triggered when:
        - the `Generate` component is rendered
        - and the queryString contains: `?addAddress=1` 
               * this enables the browser-back-button to close the modal.

_TODO:_
We still need to add all actual components to the AddAddressModal.

modal

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,6 @@ src/mocks
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+# IDE
+.idea
```
Comment (gilleswittenberg) [inline .gitignore]: Does this also work (be ignored) when you move this to your global Git config `~/.gitignore`?

Comment (Jeffrey-Zutt) [reply .gitignore]: It does! Did not know about that feature!
Added .idea to `~/.gitignore_global`

Comment (Jeffrey-Zutt) [reply .gitignore]: Dropped the commit.

File: src/components/itineraries/Generate.tsx [hunk 2]
```diff
@@ -80,6 +84,9 @@ const Generate: FC = () => {
   const showWeekDay = isWeekDay()
   const showWeekend = !showWeekDay
 
+  const { addAddress } = parseLocationSearch(window.location.search)
+  const showAddAddressModal = addAddress === '1'
```
Comment (gilleswittenberg) [inline src/components/itineraries/Generate.tsx]: Maybe more aesthetic to use hash here?

Comment (Jeffrey-Zutt) [reply src/components/itineraries/Generate.tsx]: Yeah, I agree.
Was already doubting this on implementing.

I'll refactor to use a hash here.

File: src/components/itineraries/Generate.tsx [hunk 3]
```diff
@@ -150,6 +157,14 @@ const Generate: FC = () => {
             <StyledInput type="number" value={ num } onChange={ onChangeNum } />
           </div>
         </Div>
+        <Div>
+            <Link to='?addAddress=1'>
```
Comment (gilleswittenberg) [inline src/components/itineraries/Generate.tsx]: I've been using `src/config/page.ts` to method for this to property on Links. Maybe we could extend that function to support something like `<Link to={ to(undefined, { addAddress: 1 }) } />`

File: src/components/global/Modal/DefaultModal.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+const TopBar = styled.div`
+  background-color: white;
+  position: relative;   // relative, because we want to  position the close icon absolutely within it.  
+    
+  height: ${constants.HEADER_HEIGHT_SMALL}px; 
+  text-align:center; 
+  border-bottom: 1px solid #DFDFDF;
+`
+
+// Remove default browser-styling from a button
```
Comment (Jeffrey-Zutt) [inline src/components/global/Modal/DefaultModal.tsx]: Better to use: `IconButton`

File: src/components/global/Modal/DefaultModal.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import styled from "styled-components";
+import { Close } from "@datapunt/asc-assets"
+import { constants } from "@datapunt/asc-ui"
+
+import Portal from "./Portal";
+
+const FixedWrap = styled.div`  
+  z-index: 99999; // HeaderWrap has z-index of 9999, we have top that.
```
Comment (Jeffrey-Zutt) [inline src/components/global/Modal/DefaultModal.tsx]: Add todo. We should refactor this.

File: src/components/global/Modal/DefaultModal.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    
+  overflow-y: auto;
+           
+  -webkit-overflow-scrolling: touch;  // iOS momentum scrolling. @see: https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-overflow-scrolling
+`
+const Spacing = styled.div`
+  margin: 15px;
+`
+
+interface Props {
```
Comment (Jeffrey-Zutt) [inline src/components/global/Modal/DefaultModal.tsx]: Use `type`

File: src/components/itineraries/Generate.tsx (hunk 3 above)
Comment (Jeffrey-Zutt) [reply src/components/itineraries/Generate.tsx]: Agreed to use hash.

**Pull request merged**